main/src/com/google/refine/expr/functions/arrays/Zip.java

```diff
@@ -0,0 +1,153 @@
+import com.google.refine.grel.Function;
+import com.google.refine.grel.FunctionDescription;
+
+public class Zip implements Function {
+
+    @Override
+    public Object call(Properties bindings, Object[] args) {
+        if (args.length == 1) {
+            Object[] v = (Object[]) args[0];
+            List<ArrayList> output = new ArrayList<>();
+            ArrayList arg1 = new ArrayList<>();
+            ArrayList arg2 = new ArrayList<>();
+
+            boolean isFirst = true;
+            if (v != null) {
+                for (Object o : v) {
+                    if (o.getClass().isArray() || o instanceof List<?> || o instanceof ArrayNode) {
+                        if (o.getClass().isArray()) {
+                            arg2.addAll(Arrays.asList((Object[]) o));
+                        } else if (o instanceof ArrayNode) {
+                            ArrayNode a = (ArrayNode) o;
+                            if (a.isArray()) {
+                                for (JsonNode objNode : a) {
+                                    arg2.add(objNode);
+                                }
+                            }
+                        } else {
+                            for (Object z : ExpressionUtils.toObjectList(o)) {
+                                if (z != null) {
+                                    arg2.add(z.toString());
+                                } else {
+                                    arg2.add(z);
+                                }
+                            }
+                        }
+                        if (isFirst) {
+                            for (Object x : arg2) {
+                                arg1 = new ArrayList() {
+
+                                    {
+                                        add(x);
+                                    }
+                                };
+                                output.add(arg1);
+                            }
+                        } else {
+                            output = (List<ArrayList>) Streams.zip(output.stream(),
+                                    arg2.stream(),
+                                    (a, b) -> collectElements(a, b))
+                                    .collect(Collectors.toList());
+                        }
+                        isFirst = false;
+                    }
+                    arg2.clear();
+                }
+            }
+            return output;
+        }
+        return new EvalError(EvalErrorMessage.expects_one_array(ControlFunctionRegistry.getFunctionName(this)));
+    }
+
+    @Override
+    public String getDescription() {
+        return FunctionDescription.arr_join();
+    }
+
+    @Override
+    public String getParams() {
+        return "array a, string sep";
+    }
+
+    @Override
+    public String getReturns() {
+        return "string";
+    }
+
+    private static ArrayList collectElements(Object a, Object b) {
+        ArrayList returnList = new ArrayList<>();
```

Comment:
It's good practice to use interfaces rather than concrete classes where possible.

```suggestion
    private static List collectElements(Object a, Object b) {
        List returnList = new ArrayList<>();
```